tests/mpi/periodicity_06.mpirun=2.output

```diff
@@ -0,0 +1,226 @@
+number of elements: 71
+Periodicity and hanging node constraints are consistent in parallel: 1
+=== Process 0
+Constraints:
+    0 51:  1
+    2 106:  1
+    4 53:  1
+    8 31:  1
+    11 33:  1
+    16 32:  1
+    36 9:  1
+    37 12:  1
+    40 17:  1
+    41 19:  1
+    43 23:  1
+    50 30:  1
+    52 32:  1
+    54 1:  1
+    55 5:  1
+    59 17:  1
+    60 18:  1
+    61 21:  1
+    64 44:  1
+    65 46:  1
+    66 26:  1
+    67 28:  1
+    68 30:  1
+    69 32:  1
+    71 56:  1
+    72 38:  1
+    73 17:  1
+    74 30:  1
+    75 30:  0.5
+    75 51:  0.5
+    76 30:  0.5
+    76 31:  0.5
+    77 30:  0.25
+    77 31:  0.25
+    77 51:  0.25
+    77 106:  0.25
+    78 30:  0.5
+    78 32:  0.5
+    79 30:  0.25
+    79 32:  0.25
+    79 51:  0.25
+    79 53:  0.25
+    80 30:  0.25
+    80 31:  0.25
+    80 32:  0.25
+    80 33:  0.25
+    82 51:  0.5
+    82 106:  0.5
+    83 51:  0.5
+    83 53:  0.5
+    84 6:  0.25
+    84 51:  0.25
+    84 53:  0.25
+    84 106:  0.25
+    85 31:  0.5
+    85 106:  0.5
+    86 31:  0.5
+    86 33:  0.5
+    87 6:  0.25
+    87 31:  0.25
+    87 33:  0.25
+    87 106:  0.25
+    88 6:  0.5
+    88 106:  0.5
+    89 32:  0.5
+    89 53:  0.5
+    90 32:  0.5
+    90 33:  0.5
+    91 6:  0.25
+    91 32:  0.25
+    91 33:  0.25
+    91 53:  0.25
+    92 6:  0.5
+    92 53:  0.5
+    93 6:  0.5
+    93 33:  0.5
+    103 30:  1
+    104 51:  1
+    105 31:  1
+    107 1:  1
+    108 3:  1
+    109 9:  1
+    110 10:  1
+    111 14:  1
+    114 94:  1
+    115 96:  1
+    117 100:  1
+    118 26:  1
+    119 27:  1
+    120 30:  1
+    121 31:  1
+    122 34:  1
+    123 9:  1
+    127 94:  1
+    128 95:  1
+    129 98:  1
+    130 44:  1
+    131 45:  1
+    132 48:  1
+    133 30:  1
+    134 51:  1
+    135 1:  1
+    137 124:  1
+    138 112:  1
+    139 94:  1
+    140 62:  1
+    141 44:  1
+    142 26:  1
+    143 30:  1
+Owned DoFs:
+{[0,93]}
+Ghost DoFs:
+{[0,143]}
+Coordinates:
+52@-20 20 -10
+32@20 -20 -10
+=== Process 1
+Constraints:
+    0 51:  1
+    2 106:  1
+    8 31:  1
+    32 16:  1
+    33 11:  1
+    36 9:  1
+    37 12:  1
+    40 17:  1
+    41 19:  1
+    43 23:  1
+    50 30:  1
+    52 16:  1
```

Comment:
I am not sure we are fixing the bug this unit tests used to check. There is inconsistency again, namely process 1 has
```
32 16:  1
52 16: 1
```
but process 0 has
```
16 32:  1
52 32:  1
```

so we are back to square one. I would say we still need to keep sorting DoFs when we have equality constraints. Then this together with re-arrangement of PBC vs hanging_nodes will hopefully fix the issues completely.

Reply:
p.s. fd12fdeb6c04e49d9dd6c2e2c6a546916ee82624 is not a fix of the test 😄

Reply:
We are not - as was outlined in the discussion of #7053, these constraints are not used because the dofs in question are not active. The consistency check passes...

Reply:
Ah now I understand. It is still an issue. Well easy fix.

Reply:
Pushed the sorting logic back. Running tests now.